x-pack/plugins/security/public/views/login/components/basic_login_form.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiButton, EuiCallOut, EuiFieldText, EuiFormRow, EuiPanel, EuiSpacer } from '@elastic/eui';
+import React, { ChangeEvent, Component, Fragment } from 'react';
+import { LoginState } from '../../../../common/login_state';
+
+interface Props {
+  http: any;
+  window: any;
+  infoMessage?: string;
+  loginState: LoginState;
+  next: string;
+}
+
+interface State {
+  hasError: boolean;
+  isLoading: boolean;
+  username: string;
+  password: string;
+  message: string;
+}
+
+export class BasicLoginForm extends Component<Props, State> {
+  public state = {
+    hasError: false,
+    isLoading: false,
+    username: '',
+    password: '',
+    message: '',
+  };
+
+  public componentDidMount() {
+    document.addEventListener('keydown', this.submitOnEnter);
+  }
+
+  public componentWillUnmount() {
+    document.removeEventListener('keydown', this.submitOnEnter);
+  }
+
+  public render() {
+    return (
+      <Fragment>
+        {this.renderMessage()}
+        <EuiPanel>
+          <EuiFormRow label="Username">
+            <EuiFieldText
+              name="username"
+              title="username"
+              data-test-subj="loginUsername"
+              value={this.state.username}
+              onChange={this.onUsernameChange}
+              disabled={this.state.isLoading}
+              aria-required={true}
+              inputRef={this.setUsernameInputRef}
+            />
+          </EuiFormRow>
+
+          <EuiFormRow label="Password">
+            <EuiFieldText
+              name="password"
+              title="password"
+              data-test-subj="loginPassword"
+              type="password"
+              value={this.state.password}
+              onChange={this.onPasswordChange}
+              disabled={this.state.isLoading}
+              aria-required={true}
+            />
+          </EuiFormRow>
+
+          <EuiSpacer />
+          <EuiButton
+            fill
+            color="primary"
+            onClick={this.submit}
+            isLoading={this.state.isLoading}
+            isDisabled={!this.isFormValid()}
+            data-test-subj="loginSubmit"
+          >
+            Log in
+          </EuiButton>
+        </EuiPanel>
+      </Fragment>
+    );
+  }
+
+  private renderMessage = () => {
+    if (this.state.message) {
+      return (
+        <Fragment>
+          <EuiCallOut
+            size="s"
```

Comment:
Could we please add `role="alert"` to this element. That will highly increase the accessibility of this error message, since it will automatically be read out to a screenreader user, once it appears.

Reply:
Ups, just saw @snide already wrote a comment about that in the PR, sorry :D